src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -55,6 +56,14 @@ public void locateComponentsForOfflineDetectorScan(BdioResult bdio, File scanOut
     public void locateComponentsForOnlineIntelligentScan() throws ComponentLocatorException {
         logger.info(ReportConstants.RUN_SEPARATOR);
         logger.info("Intelligent Scan mode does not support Component Location Analysis.");
+        failComponentLocationAnalysisOperation();
+    }
+
+    /**
+     * A ComponentLocatorException should be thrown in all cases where the output file is not created so that the
+     * appropriate status can be logged for the requested {@link GenerateComponentLocationAnalysisOperation}.
+     */
+    public void failComponentLocationAnalysisOperation() throws ComponentLocatorException {
         throw new ComponentLocatorException("Failed to generate Component Location Analysis file.");
```

Comment:
Nit: Can we pass on a specific message to identify the case when throwing this message than just this generic message?

Reply:
We communicate to the user through INFO level (or higher) logs what the specific reason was (for example unsupported scan mode, problem during locator execution, dummy lib hit, etc). This exception itself is only reported under "Detect Exceptions" as far as the console output is concerned.  

Taking from the ComponentLocatorException class' description: 
"The ComponentLocatorException is needed to appropriately set and report the status of Component Locator related operations while letting Detect exit successfully in all cases."

In the future we can have more specific information from CLL on the reason for a failure, maybe even implement different exit codes, etc and at that point the exception can be more informative.